Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,253 @@
+		TransportData = 4         ///< Transport Data message
+	};
+
+	/**
+	 * @struct wg_common_header
+	 * Represents the common header for all WireGuard message types
+	 */
+#pragma pack(push, 1)
+	struct wg_common_header
+	{
+		/** Message type field */
+		uint8_t messageType;
+		/** Reserved field (3 bytes) */
+		uint8_t reserved[3];
+	};
+#pragma pack(pop)
+
+	/**
+	 * @struct wg_handshake_initiation
+	 * Represents the Handshake Initiation message
+	 */
+#pragma pack(push, 1)
+	struct wg_handshake_initiation
+	{
+		wg_common_header common;               ///< Common header for all WireGuard messages
+		uint32_t senderIndex;                  ///< Sender index
+		uint8_t initiatorEphemeral[32];        ///< Initiator's ephemeral public key
+		uint8_t encryptedInitiatorStatic[48];  ///< Encrypted initiator's static key
+		uint8_t encryptedTimestamp[28];        ///< Encrypted timestamp
+		uint8_t mac1[16];                      ///< MAC1 field
+		uint8_t mac2[16];                      ///< MAC2 field
+	};
+#pragma pack(pop)
+
+	/**
+	 * @struct wg_handshake_response
+	 * Represents the Handshake Response message
+	 */
+#pragma pack(push, 1)
+	struct wg_handshake_response
+	{
+		wg_common_header common;         ///< Common header for all WireGuard messages
+		uint32_t senderIndex;            ///< Sender index
+		uint32_t receiverIndex;          ///< Receiver index
+		uint8_t responderEphemeral[32];  ///< Responder's ephemeral public key
+		uint8_t encryptedEmpty[16];      ///< Encrypted empty field
+		uint8_t mac1[16];                ///< MAC1 field
+		uint8_t mac2[16];                ///< MAC2 field
+	};
+#pragma pack(pop)
+
+	/**
+	 * @struct wg_cookie_reply
+	 * Represents the Cookie Reply message
+	 */
+#pragma pack(push, 1)
+	struct wg_cookie_reply
+	{
+		wg_common_header common;      ///< Common header for all WireGuard messages
+		uint32_t receiverIndex;       ///< Receiver index
+		uint8_t nonce[24];            ///< Nonce field
+		uint8_t encryptedCookie[32];  ///< Encrypted cookie
+	};
+#pragma pack(pop)
+
+	/**
+	 * @struct wg_transport_data
+	 * Represents the Transport Data message
+	 */
+#pragma pack(push, 1)
+	struct wg_transport_data
+	{
+		wg_common_header common;   ///< Common header for all WireGuard messages
+		uint32_t receiverIndex;    ///< Receiver index
+		uint64_t counter;          ///< Counter field
+		uint8_t encryptedData[0];  ///< Flexible array member for encrypted data
+	};
+#pragma pack(pop)
+
+	/**
+	 * @class WireGuardLayer
+	 * Represents a WireGuard protocol layer
+	 */
+	class WireGuardLayer : public Layer
+	{
+	public:
+		/**
+		 * Constructs a WireGuardLayer object.
+		 *
+		 * @param data Pointer to the raw data representing the WireGuard layer
+		 * @param dataLen Length of the data
+		 * @param prevLayer Pointer to the previous layer in the packet (if any)
+		 * @param packet Pointer to the packet this layer belongs to
+		 */
+		WireGuardLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
```

Comment:
It'd be nice to be able to create WG packets also and not only parse them. Here we also have 2 options:
1. If we create a class per message type, it's easy because each class will have its own c'tor
2. If we decide to keep one `WireGuardLayer` class we can do what we do in [`IcmpLayer`](https://github.com/seladb/PcapPlusPlus/blob/master/Packet%2B%2B/header/IcmpLayer.h) which is add an empty c'tor, and have setter methods for each layer type which extend or shorten the layer and then set the correct data

Reply:
I would like to propose option 3, which is tagged constructors. Basically we have a special structure thàt is the MessageTag, that is passed as first param to the constructor to differentiate which overloads are to be used even if the overloads from different messages use similar input parameters.

Reply:
@seladb I've implemented the feature to create WireGuard messages using constructors, following option 1.

@Dimi1010 Thank you for your suggestion regarding tagged constructors. However, I have already implemented the functionality using option 1. If any additional modifications are needed, I will be happy to consider them.